param.go

```diff
@@ -395,7 +395,22 @@ func newParamObject(t reflect.Type, c containerStore) (paramObject, error) {
 
 func (po paramObject) Build(c containerStore) (reflect.Value, error) {
 	dest := reflect.New(po.Type).Elem()
+	var groupQueue []paramObjectField
 	for _, f := range po.Fields {
+		p, ok := f.Param.(paramGroupedSlice)
+		if ok {
+			if p.Soft {
+				groupQueue = append(groupQueue, f)
+				continue
+			}
+		}
+		v, err := f.Build(c)
+		if err != nil {
+			return dest, err
+		}
+		dest.Field(f.FieldIndex).Set(v)
```

Comment:
looks like there's some amount of dup code here. i think we can simplify this block to reduce some complexity -- you're essentially re-organizing the ordering of the fields so that you build non-groups first. Maybe just build out two queues first by iterating through the whole fields and then merge the queues to actually call Build on them.

Also you might want to add a comment on why this is necessary as it's not immediately obvious why we need to do this.